src/components/Admin/StudentGroupPanel/index.tsx

```diff
@@ -4,41 +4,126 @@ import styles from './styles.module.scss';
 import { observer } from 'mobx-react-lite';
 import { useStore } from '@tdev-hooks/useStore';
 import Button from '@tdev-components/shared/Button';
-import { mdiPlusCircleOutline } from '@mdi/js';
+import { mdiMagnify, mdiPlusCircleOutline, mdiRestore } from '@mdi/js';
 import StudentGroup from '@tdev-components/StudentGroup';
 import _ from 'es-toolkit/compat';
-import scheduleMicrotask from '@tdev-components/util/scheduleMicrotask';
 import { action } from 'mobx';
+import Icon from '@mdi/react';
 
 const StudentGroupPanel = observer(() => {
     const userStore = useStore('userStore');
     const groupStore = useStore('studentGroupStore');
     const current = userStore.current;
+    const [searchFilter, setSearchFilter] = React.useState('');
+    const [searchRegex, setSearchRegex] = React.useState(new RegExp(searchFilter, 'i'));
+
+    React.useEffect(() => {
+        setSearchRegex(new RegExp(searchFilter, 'i'));
+    }, [searchFilter]);
+
     if (!current?.hasElevatedAccess) {
         return null;
     }
     return (
         <div>
-            <Button
-                onClick={() => {
-                    groupStore.create('', '').then(
-                        action((group) => {
-                            group?.setEditing(true);
-                        })
-                    );
-                }}
-                icon={mdiPlusCircleOutline}
-                color="primary"
-                text="Neue Lerngruppe erstellen"
-            />
+            <div className={clsx(styles.controls)}>
+                <Button
+                    onClick={() => {
+                        groupStore.create('', '').then(
+                            action((group) => {
+                                group?.setEditing(true);
+                            })
+                        );
+                    }}
+                    icon={mdiPlusCircleOutline}
+                    color="primary"
+                    text="Neue Lerngruppe erstellen"
+                />
+                <div className={clsx(styles.searchBox)}>
+                    <Icon path={mdiMagnify} size={1} />
+                    <div className={clsx(styles.searchInput)}>
+                        <input
+                            type="text"
+                            placeholder="Lerngruppen filtern..."
+                            value={searchFilter}
+                            className={clsx(styles.textInput)}
+                            onChange={(e) => {
+                                setSearchFilter(e.target.value);
+                            }}
+                        />
+                        <Button
+                            onClick={() => {
+                                setSearchFilter('');
+                            }}
+                            icon={mdiRestore}
```

Comment:
just wondering, isn't the default icon to reset search filters a "x"? e.g. `mdiClose`?